identity-account/src/account/builder.rs

```diff
@@ -79,48 +96,104 @@ impl AccountBuilder {
+      Some(AccountStorage::Memory) => {
+        let storage = Arc::new(MemStore::new());
+        self.storage = Some(storage);
+      }
       #[cfg(feature = "stronghold")]
-      AccountStorage::Stronghold(snapshot, password) => {
+      Some(AccountStorage::Stronghold(snapshot, password)) => {
         let passref: Option<&str> = password.as_deref();
         let adapter: Stronghold = Stronghold::new(&snapshot, passref).await?;
 
         if let Some(mut password) = password {
           password.zeroize();
         }
 
-        Account::with_config(adapter, self.config).await?
+        let storage = Arc::new(adapter);
+        self.storage = Some(storage);
       }
-      AccountStorage::Custom(adapter) => Account::with_config(adapter, self.config).await?,
+      Some(AccountStorage::Custom(storage)) => {
+        self.storage = Some(storage);
+      }
+      None => (),
     };
 
-    if let Some(clients) = self.clients.take() {
-      for (_, client) in clients.into_iter() {
-        account.set_client(client.build().await?);
+    // unwrap is fine, since by default, storage_template is `Some`,
+    // which results in storage being `Some`.
+    // Overwriting storage_template always produces `Some` storage.
+    Ok(Arc::clone(self.storage.as_ref().unwrap()))
+  }
+
+  /// Apply configuration to the IOTA Tangle client for the given [`Network`].
+  pub fn client<F>(mut self, network: Network, f: F) -> Self
+  where
+    F: FnOnce(ClientBuilder) -> ClientBuilder,
+  {
+    self
+      .client_builders
+      .get_or_insert_with(HashMap::new)
+      .insert(network.name(), f(ClientBuilder::new().network(network)));
+    self
+  }
+
+  async fn build_clients(&mut self) -> Result<()> {
+    if let Some(hmap) = self.client_builders.take() {
+      for builder in hmap.into_iter() {
+        self.client_map.insert(builder.1.build().await?)
       }
     }
 
+    Ok(())
+  }
+
+  /// Creates a new identity based on the builder configuration and returns
+  /// an [`Account`] instance to manage it.
+  /// The identity is stored locally in the [`Storage`].
+  ///
+  /// See [`IdentityCreate`] to customize the identity creation.
+  pub async fn create_identity(&mut self, input: IdentitySetup) -> Result<Account> {
+    self.build_clients().await?;
+
+    let setup = AccountSetup::new_with_options(
+      self.get_storage().await?,
+      Some(self.config.clone()),
+      Some(Arc::clone(&self.client_map)),
+    );
+
+    let account = Account::create_identity(setup, input).await?;
+
+    Ok(account)
+  }
+
+  /// Loads an existing identity with the specified `did` using the current builder configuration.
+  /// The identity must exist in the configured [`Storage`].
+  pub async fn load_identity(&mut self, did: IotaDID) -> Result<Account> {
+    self.build_clients().await?;
+
+    let setup = AccountSetup::new_with_options(
+      self.get_storage().await?,
+      Some(self.config.clone()),
+      Some(Arc::clone(&self.client_map)),
+    );
```

Comment:
> I think the way around this would be to clone the client_builders each time to create a brand new ClientMap?

The `client_builders` are added to the `client_map` eventually, so to avoid modifying older `ClientMap`s we would have to clone the `client_map` before constructing each `Account`. This duplication is what the current code is trying to avoid. I think the current trade-off is better than preventing some potential edge cases at the cost of higher memory usage.

Reply:
> I think the current trade-off is better than preventing some potential edge cases at the cost of higher memory usage.

I'm tempted to say the extra memory of cloning the `ClientMap` is insignificant but I have no data to back that up, and memory usage is an issue for the Wasm bindings. I think the way forward is to clearly document that, similar to the `Storage`, the `ClientMap` is shared between `Account` instances generated from the same `AccountBuilder` and that methods like `set_client` will update it.

Reply:
For future reference: Due to the large size of the Client (1248 bytes), we decided to continue with the current `Arc` approach.